python/pysrc/kaskada/sources/source.py

```diff
@@ -3,13 +3,16 @@
 
 import kaskada._ffi as _ffi
 import pyarrow as pa
+from typing import Literal
+from typing_extensions import TypeAlias
 
 from .._session import _get_session
 from .._timestream import Timestream
 
 
 _TABLE_NUM: int = 0
 
+TimeUnit: TypeAlias = Optional[Literal["ns", "us", "ms", "s"]]
```

Comment:
I think I'd prefer removing the `Optional` here and explicitly writing `Optional[TimeUnit]` where needed, so we can clearly see in the signature whether it's required. I acknowledge that it's generally got a default value to the parameters where it's used, so not a big deal.

Reply:
I like that. Making that change.